packages/jupyter-ai/src/components/settings/model-parameters-input.tsx

```diff
@@ -10,22 +20,36 @@ type ModelParameter = {
   isStatic?: boolean;
 };
 
-type StaticParameterDef = {
-  name: string;
-  type: string;
-  label: string;
+export type ModelParametersInputProps = {
+  modelId?: string | null;
 };
 
-// Add some common fields as static parameters here
-const STATIC_PARAMETERS: StaticParameterDef[] = [
-  { name: 'temperature', type: 'float', label: 'Temperature' },
-  { name: 'api_url', type: 'string', label: 'API URL' },
-  { name: 'max_tokens', type: 'integer', label: 'Max Tokens' }
-];
-
-export function ModelParametersInput(): JSX.Element {
+export function ModelParametersInput(
+  props: ModelParametersInputProps
+): JSX.Element {
+  const [availableParameters, setAvailableParameters] = useState<any>(null);
   const [parameters, setParameters] = useState<ModelParameter[]>([]);
   const [validationError, setValidationError] = useState<string>('');
+  const alert = useStackingAlert();
+
+  useEffect(() => {
+    async function fetchAvailableParameters() {
+      if (!props.modelId) {
+        setAvailableParameters(null);
+        return;
+      }
+
+      try {
+        const response = await AiService.getModelParameters(props.modelId);
```

Comment:
No indication of loading state while fetching won't allow users to to distinguish loading vs empty state. See `model-id-input.tsx` and  `secrets-section.tsx` for examples of loading state implementation.

Reply:
Good callout. I've opened a follow-up issue for this: #1463.